### PR TITLE
feat(analytics): weekly outfit variety score (#18 follow-up)

### PR DESCRIPTION
**Pull request by William-Hill**
## Summary

Closes the remaining "Weekly outfit variety score in analytics" bullet from #18 (PR #210 landed the suggestion-time repeat detection half).

- **`server/services/variety.py`** — `compute_variety_score(client, user_id, days=7)` returns `{score, days, outfits_logged, unique_core_items, repeated_combinations}`. Score = `1 - repeated_pairs / max_pairs`, where two wear events count as a repeat when their core (Tops/Bottoms/Footwear) item-id overlap is `>= 2`. Same threshold used by `_annotate_with_wear_history` in `outfit.py`. Insufficient data (`< 2` outfits) returns `1.0` so new users aren't penalized.
- **`server/services/feedback_context.py`** — `get_recent_wear_history` now accepts an optional `include_items=True` kwarg that adds `items: [{id, garment_category}]` to each entry, so the variety scorer can filter to core items without a re-query. Default `include_items=False` keeps the existing call site in `routers/outfit.py` byte-identical.
- **`server/routers/analytics.py`** — new `GET /analytics/variety` endpoint, mounted in both `server/main.py` and `entrypoint.py` (matching the outfit router pattern).
- **Mobile** — `mobile/src/lib/analytics/queries.ts` (`fetchVarietyScore`) calls the new endpoint with the same `EXPO_PUBLIC_AGENT_API_URL` + bearer-token pattern as `chat/api.ts`. `mobile/src/components/analytics/variety-score-card.tsx` renders the score with FONTS.serif for the number and FONTS.mono for labels, matching the editorial aesthetic of `outfit-card-v2.tsx`. Mounted on the profile screen above the action list.
- **Tests** — `tests/test_variety.py` covers: insufficient data, no data, full overlap on every pair (score 0), all-distinct (score 1), partial / mixed overlap, accessory-only overlap (not a repeat), and single-core-item overlap (not a repeat).

### Decisions made

- Extended `get_recent_wear_history` with an opt-in `include_items=True` kwarg rather than adding a sibling function. Smaller change, keeps the original return shape default, and avoids duplicating the date-filter / row-loop logic.
- Score is displayed as a percentage in the UI (e.g. `83%`) because the score itself is `0..1` and a percent reads more intuitively at a glance. The raw float is still in the API payload for any future analytics consumer.
- Card lives above the action list on the profile screen — small visible win without burying it.

## Test plan

- [x] `python -m pytest tests/test_variety.py tests/test_feedback_context.py tests/test_outfit_endpoint.py tests/test_outfit_models.py tests/test_outfit_repeat_detection.py -q` — 47 passed
- [ ] Manual: log >= 2 `wore_it` events in the app, hit `GET /analytics/variety`, confirm payload
- [ ] Manual: open Profile tab, verify the card renders, shows "insufficient data" copy when < 2 outfits logged
- [ ] Manual: confirm accessory overlap (same hat across days) does not lower the score

## Summary by CodeRabbit

* **New Features**
  * Outfit variety analytics added to track wardrobe diversity (weekly).
  * Variety Score card added to the mobile profile and prompts users to log more outfits when data is limited.
  * Server now exposes analytics endpoints so variety metrics are available to the app.

* **Tests**
  * Added comprehensive test suite covering variety score calculations, edge cases, and legacy data handling.

[Review Change Stack](https://app.coderabbit.ai/change-stack/William-Hill/ptso_agent/pull/211)

### CODE REVIEW

File: server/services/variety.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    items = entry.get("items")
+    if isinstance(items, list) and items:
+        return {
+            str(it.get("id"))
+            for it in items
+            if isinstance(it, dict)
+            and it.get("id")
+            and it.get("garment_category") in _CORE_CATEGORIES
+        }
+    return set()
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply server/services/variety.py:51]: Fixed in 1931b58 — `_core_ids` now detects when no item has `garment_category` set and falls back to `entry["item_ids"]` as the docstring promised. Added `test_legacy_entries_fall_back_to_item_ids` covering the regression.

File: server/services/variety.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply server/services/variety.py:52]: Fixed in 1931b58 — same fix as the Codex thread above: `_core_ids` falls back to `item_ids` when no items carry category metadata. Test added.

File: mobile/src/components/analytics/variety-score-card.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+import { useThemeTokens } from '@/src/lib/theme/tokens';
+import { FONTS } from '@/src/components/chat/editorial-fonts';
+import {
+  fetchVarietyScore,
+  type VarietyScore,
+} from '@/src/lib/analytics/queries';
+
+function formatPercent(score: number): string {
+  const clamped = Math.max(0, Math.min(1, score));
+  return `${Math.round(clamped * 100)}%`;
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply mobile/src/components/analytics/variety-score-card.tsx:14]: Fixed in 1931b58 — `formatPercent` now coerces non-finite scores to 0 before clamping.

File: mobile/src/components/analytics/variety-score-card.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        borderColor: t.borderDefault,
+        backgroundColor: t.surface,
+      }}
+    >
+      <Text
+        variant="overline"
+        style={{ color: t.textMuted, letterSpacing: 1.8 }}
+      >
+        VARIETY · LAST 7 DAYS
+      </Text>
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply mobile/src/components/analytics/variety-score-card.tsx:63]: Fixed in e12e08d — heading now reads `VARIETY · LAST ${data?.days ?? 7} DAYS`, sourced from the payload.

File: server/services/variety.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      score: float in [0, 1] — higher = more variety. 1.0 when there's
+        insufficient data (< 2 outfits logged).
+      days: window used.
+      outfits_logged: number of `wore_it` events in window.
+      unique_core_items: distinct top/bottom/footwear item ids worn.
+      repeated_combinations: pairs of wear events with core overlap >=
+        `_REPEAT_OVERLAP_THRESHOLD`.
+    """
+    history = get_recent_wear_history(client, user_id, days=days, include_items=True)
+    outfits_logged = len(history)
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply server/services/variety.py:70]: Fixed in e12e08d — `compute_variety_score` now clamps with `days = max(1, int(days))` (chose silent coercion over `ValueError` to match the lenient input handling elsewhere in the codebase). Added `test_days_coerced_to_minimum_of_1`.

File: mobile/src/components/analytics/variety-score-card.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        borderColor: t.borderDefault,
+        backgroundColor: t.surface,
+      }}
+    >
+      <Text
+        variant="overline"
+        style={{ color: t.textMuted, letterSpacing: 1.8 }}
+      >
+        {`VARIETY · LAST ${data?.days ?? 7} DAYS`}
+      </Text>
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply mobile/src/components/analytics/variety-score-card.tsx]: Fixed in d7d4261 — both the eyebrow heading and the subtitle now switch between DAY/DAYS and day/days based on `data.days === 1`.

File: tests/test_variety.py [hunk 7]
```diff
@@ -0,0 +1,184 @@
+"""Tests for the weekly outfit variety score service."""
+
+from datetime import datetime, timedelta, timezone
+from unittest.mock import MagicMock
+
+from server.services.variety import compute_variety_score
+
+
+def _wear_row(items, hours_ago):
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply tests/test_variety.py]: Fixed in d7d4261 — added `list[tuple[str, str]] -> dict[str, Any]`, `list[dict[str, Any]] -> MagicMock`, and `list[str] -> dict[str, Any]` return types on the three private helpers (ANN202 cleared).
